### PR TITLE
Fix hanging terminal editor during conflict resolution

### LINKED ISSUE #499: providing terminal editor during conflicts hangs the process
when providing editor like nano or (n)vim
during a cherry pick conflict resolution I never get to the listing of files
to resolve

it hangs one step before.

image

And waits for a stdin input, which is understandable because nvim as any terminal editor is operating in interactive mode
and expecting stdin to be continuously fed.

Since `child_process.spawn`s default `stdio` option is `pipe` it creates an indirection layer between parent and child pipes, making nvim or any other editor hang disconnected while waiting for any input.

One way of fixing this would be to add `stdio: inherit` to `spawn`, to remove the unnecessary pipe layer and connect child process to parent io pipes.

I already tested it and it indeed works. Let me know wdyt.

Thanks

### PR DESCRIPTION
**Pull request by kapral18**
Fixes #499

### CODE REVIEW

File: src/lib/child-process-promisified.ts [hunk 1]
```diff
@@ -36,6 +36,7 @@ export async function spawnPromise(
   cmd: string,
   cmdArgs: ReadonlyArray<string>,
   cwd: string,
+  isInteractive = false,
```
Comment (sorenlouv) [inline src/lib/child-process-promisified.ts:39]: Would `inheritOutput` perhaps be better?

```suggestion
  inheritOutput = false,
```

Comment (kapral18) [reply src/lib/child-process-promisified.ts:39]: The reasoning behind my suggestion:

`isInteractive` is my attempt to abstract away from details of implementation and focus on behavior. if tomorrow node api will change `stdio` option and to achieve interactive process you would need some other combination of stdio values or even other name you wouldn't need to change the name of your flag for other consumers inside your codebase or maybe outside if you decide to export this api.

But other than that fine with me. Technically it inherits not only output but also input, so we could also rename it to `inheritStdio`.

But ultimately up to you.

Comment (sorenlouv) [reply src/lib/child-process-promisified.ts:39]: Good point. Let's just leave this as-is.

**Review by sorenlouv — APPROVED** [review]
Looks good. I'll take this for a manual test later today, then merge if all is good. Thanks for the contribution

**Comment by sorenlouv** [comment]
Released in v9.5.1

**Pull request merged**